SnapshotTests/ThemeRemoteConfigurationTests.swift

```diff
@@ -0,0 +1,34 @@
+@testable import GliaWidgets
+import XCTest
+import SnapshotTesting
+
+final class ThemeRemoteConfigurationTests: SnapshotTestCase {
+    func test_themeForMessagesFromHistory() throws {
+        let url = try XCTUnwrap(
+            Bundle(for: Self.self)
+                .url(
+                    forResource: "unified_sample.json",
+                    withExtension: nil
+                )
+        )
+
+        let jsonData = try Data(contentsOf: url)
+        let config = try JSONDecoder().decode(
+            RemoteConfiguration.self,
+            from: .init(jsonData)
+        )
+
+        let theme = Theme(
+            uiConfig: config,
+            assetsBuilder: .standard
+        )
+
+        let viewController = ChatViewController.mockHistoryMessagesScreen(theme: theme)
+        viewController.view.frame = UIScreen.main.bounds
+        assertSnapshot(
+            matching: viewController,
+            as: .image,
+            named: nameForDevice()
+        )
```

Comment:
Need confirmation that I understood that correctly..
Tests generate an image of the mocked view that is generated based on remote theme configurations`unified_sample` and compare it to the view that is currently on the device screen? Is that correct?

Reply:
But basically, the test is using the same view as the Widget SDK for the test but just with mocked properties? Wouldn't that mean that if there is a bug in setting the color of the icon then whatever the theme (default/remote) the result would be the same and the test would anyway pass?

Reply:
@gugalo Using same view with mocked properties will not pass if theme related configuration logic will change, because snapshots will be different. 

Test scenario works like this:
1. Configure theme using `unified_sample.json`.
2. Apply configured theme to UI.
3. Bring UI into desired state by mocking business logic.
4. Create snapshot image of a view under test.
5. Compare current snapshot with previously generated one.
6. If snapshots ([current](
https://user-images.githubusercontent.com/3148347/242498686-ef225be3-54b2-4228-bc41-9f59a2ac2d0f.png) and [previous](https://user-images.githubusercontent.com/3148347/242496791-79b274e0-fa18-42fd-bb54-f0f779eb11a8.png)) have difference, then raise test failure, resulting in [diff image](
https://user-images.githubusercontent.com/3148347/242500384-f93afdea-c0b2-4622-b816-6d652fd07d8e.png)
7. In case this test is run for the first time ever, it will generate initial image, but it will also fail, forcing test creator to check if generated image looks correctly. That initial generated image will serve as reference for the subsequent test cases.

Reply:
Thank you! That was very nice explanation 👍